src/python/ensembl/io/genomio/genome_metadata/prepare.py

```diff
@@ -75,89 +75,87 @@ class MetadataError(Exception):
+    Assembly provider metadata will only be added if it is missing, i.e. neither `"provider_name"` or
+    `"provider_url"` are present. The gene model metadata will only be added if `gff3_file` is provided.
 
     Args:
         genome_data: Genome information of assembly, accession and annotation.
         gff3_file: Path to GFF3 file to use as annotation source for this genome.
 
+    Raises:
+        MetadataError: If accession's format in genome metadata does not match with a known provider.
     """
     # Get accession provider
-    accession = genome_data["assembly"]["accession"]
+    accession = genome_metadata["assembly"]["accession"]
     if accession.startswith("GCF"):
         provider = PROVIDER_DATA["RefSeq"]
     elif accession.startswith("GCA"):
         provider = PROVIDER_DATA["GenBank"]
     else:
-        raise MetadataError(f"Accession doesn't look like an INSDC or RefSeq accession: {accession}")
+        raise MetadataError(f"Accession does not look like an INSDC or RefSeq accession: {accession}")
 
     # Add assembly provider (if missing)
-    assembly = genome_data["assembly"]
+    assembly = genome_metadata["assembly"]
     if (not "provider_name" in assembly) and (not "provider_url" in assembly):
         assembly["provider_name"] = provider["assembly"]["provider_name"]
-        assembly["provider_url"] = provider["assembly"]["provider_url"]
+        assembly["provider_url"] = f'{provider["assembly"]["provider_url"]}/{accession}'
 
     # Add annotation provider if there are gene models
     if gff3_file:
-        annotation = {}
-        if "annotation" in genome_data:
-            annotation = genome_data["annotation"]
+        annotation = genome_metadata.setdefault("annotation", {})
         if ("provider_name" not in annotation) and ("provider_url" not in annotation):
             annotation["provider_name"] = provider["annotation"]["provider_name"]
-            annotation["provider_url"] = provider["annotation"]["provider_url"]
-        genome_data["annotation"] = annotation
+            annotation["provider_url"] = f'{provider["annotation"]["provider_url"]}/{accession}'
 
 
 def add_assembly_version(genome_data: Dict) -> None:
-    """Adds version number to the genome's assembly if one is not present already.
+    """Adds version number to the genome's assembly information if one is not present already.
 
     Args:
         genome_data: Genome information of assembly, accession and annotation.
-
     """
     assembly = genome_data["assembly"]
     if not "version" in assembly:
         accession = assembly["accession"]
-        values = accession.split(".")
-        if (len(values) == 2) and values[1]:
-            assembly["version"] = int(values[1])
+        version = accession.partition(".")[2]
+        if version:
+            assembly["version"] = int(version)
 
 
 def add_genebuild_metadata(genome_data: Dict) -> None:
```

Comment:
Perhaps use a bit more descriptive function name here. Perhaps: `add_genebuild_version_metadata`

Reply:
This function is adding both `version` and `start_date`, so I think in this case the function name is appropriate. That being said, maybe referring to annotation rather than genebuild would be more accurate, but that can be discussed separaterly.